### PR TITLE
Updated SPH Sample Inputs

### PR DESCRIPTION
**Pull request by wbinventor**
This short PR updates the SPH examples based on mit-crpg/openmc#644. In addition, the `openmoc.opencg_compatible` module now includes periodic boundary conditions.

### CODE REVIEW

File: sample-input/sph-factors/pin-cell/sph-factors.py [hunk 1]
```diff
@@ -60,17 +55,16 @@
 
 # Compute SPH factors
 sph, sph_mgxs_lib, sph_indices = \
-    openmoc.materialize.compute_sph_factors(mgxs_lib,
-                                            azim_spacing=opts.azim_spacing,
-                                            num_azim=opts.num_azim,
-                                            num_threads=opts.num_omp_threads)
+    openmoc.materialize.compute_sph_factors(
+        mgxs_lib, azim_spacing=opts.azim_spacing,
+        num_azim=opts.num_azim, num_threads=opts.num_omp_threads)
```
Comment (geogunow) [inline sample-input/sph-factors/pin-cell/sph-factors.py]: The spacing looks a bit weird here

File: sample-input/sph-factors/slab/sph-factors.py [hunk 2]
```diff
@@ -60,17 +55,16 @@
 
 # Compute SPH factors
 sph, sph_mgxs_lib, sph_indices = \
-    openmoc.materialize.compute_sph_factors(mgxs_lib,
-                                            azim_spacing=opts.azim_spacing,
-                                            num_azim=opts.num_azim,
-                                            num_threads=opts.num_omp_threads)
+    openmoc.materialize.compute_sph_factors(
+        mgxs_lib, azim_spacing=opts.azim_spacing,
+        num_azim=opts.num_azim, num_threads=opts.num_omp_threads)
```
Comment (geogunow) [inline sample-input/sph-factors/slab/sph-factors.py]: ditto

**Comment by geogunow** [comment]
Aside from a few things, everything looks good here. Although I should point out I know nothing about OpenMC syntax so I'm assuming the compatibility is fine.

File: sample-input/sph-factors/pin-cell/sph-factors.py (hunk 1 above)
Comment (wbinventor) [inline sample-input/sph-factors/pin-cell/sph-factors.py]: Yeah it's unlike indentation elsewhere in the code but is PEP8 compliant

**Pull request merged**